pkg/runner/test_runner_pipeline.go

```diff
@@ -198,174 +207,161 @@ func (p *pipeline) runStep(cancel, pause <-chan struct{}, jobID types.JobID, run
 	}
 }
 
-// waitTargets reads results coming from results channels until all Targets
-// have completed or an error occurs. If all Targets complete successfully, it checks
-// whether Steps and routing blocks have completed as well. If not, returns an
-// error. Termination is signalled via terminate channel.
-func (p *pipeline) waitTargets(terminate <-chan struct{}, completedCh chan<- *target.Target) error {
+// emitStepEvent emits a failure event if a step fail
+func (p *pipeline) emitStepEvent(result *stepResult) error {
```

Comment:
I interpret this name differently from what it does: I interpret that it sends an event on either success or failure, while actually it sends an event only on failure. BTW, why we do not send successes? To reduce amount of rows in DB?